flink-yarn/src/main/java/org/apache/flink/yarn/YarnClusterDescriptor.java

```diff
@@ -682,6 +683,10 @@ private ApplicationReport startAppMaster(
 				// add user code jars from the provided JobGraph
 				: jobGraph.getUserJars().stream().map(f -> f.toUri()).map(File::new).collect(Collectors.toSet());
 
+		int yarnFileReplication = yarnConfiguration.getInt(DFSConfigKeys.DFS_REPLICATION_KEY, DFSConfigKeys.DFS_REPLICATION_DEFAULT);
+		int fileReplication = flinkConfiguration.getInteger(YarnConfigOptions.FILE_REPLICATION);
```

Comment:
what if user configured `YarnConfigOptions.FILE_REPLICATION` with `0`?

Reply:
I change the condition to  fileReplication = fileReplication > 0 ? fileReplication : yarnFileReplication;

Reply:
Do you have a test to update the `DFS_REPLICATION_KEY` in hdfs-site.xml? And then the replication here should be same. I am not sure we could get a updated value from `yarnConfiguration`. Since `YarnConfiguration` do not add the `hdfs-site.xml` as resource by default.